### PR TITLE
VSCode capture

### PR DESCRIPTION
**Pull request by Daanikus**

### CODE REVIEW

File: vscode/src/capture/command.ts [hunk 1]
```diff
@@ -32,19 +43,29 @@ async function storeRule(message: string, source: CaptureSource) {
...
     async () => {
-      // TODO: store rule  here using a codelingo API
-      // const rule = await axios.post(`${config.apiEndpoint}/rules/add/...`, {
-      //   ...
-      // });
-      // ...
-      //
+      const api = config.api;
+
+      const response = await axios.post(
```
Comment (Daanikus) [inline vscode/src/capture/command.ts:51]: Gotta handle these errors but at least VSCode displays an error msg if things go wrong here

File: vscode/src/ui.ts [hunk 2]
```diff
@@ -7,7 +8,7 @@ export async function showRuleWasCreated(rule: Rule, source: CaptureSource) {
     'View Rule'
   );
   if (response === 'View Rule') {
-    const url = workspace.getConfiguration('codelingo').get<string>('launchUrl')!;
+    const url = `${config.dash.host}/repos/${source.owner}/${source.repo}/rules/${rule.id}`;
```
Comment (Daanikus) [inline vscode/src/ui.ts:11]: TODO: change URL based on prod/staging

File: vscode/src/ui.ts [hunk 3]
```diff
@@ -61,6 +62,17 @@ export async function errorNoRepoDetected() {
   await window.showErrorMessage('Could not detect which GitHub repo this file belongs to');
 }
 
+export async function errorRepoNotFound() {
+  await window.showErrorMessage(
+    `CodeLingo was unable to find your repo. Please make sure the CodeLingo GitHub app is installed
+    on GitHub.`
```
Comment (rstorr) [inline vscode/src/ui.ts:68]: quibble but 'Please make sure the CodeLingo GitHub app is installed' - ha obviously app is on GH

File: vscode/src/capture/command.ts [hunk 4]
```diff
@@ -32,19 +43,29 @@ async function storeRule(message: string, source: CaptureSource) {
...
+        `${api.host}/${api.paths.capture}/${source.owner}/${source.repo}`,
+        {
+          name: message,
+          description: descriptionFromSource(source),
+          query: defaultQuery,
+          functions: null,
+          review_comment: '---',
```
Comment (rstorr) [inline vscode/src/capture/command.ts:58]: review_comment can't be null? maybe 'insert review comment'/'default review comment' would be better here?

what's the default when we capture from the dash?

File: vscode/src/capture/command.ts [hunk 5]
```diff
@@ -71,5 +93,13 @@ async function inferContextFromActiveEditor(): Promise<{
...
-  return { repos, filepath, lineRange };
+  const commit = await repo.getCommit('HEAD');
+
+  return { repos, filepath, lineRange, commit };
+}
+
+function descriptionFromSource(source: CaptureSource) {
+  return `Rule captured from lines ${source.lineRange?.[0]} to ${source.lineRange?.[1]} of ${
+    source.filepath
+  } in commit ${source.commit?.hash.substring(0, 7)}`;
```
Comment (rstorr) [inline vscode/src/capture/command.ts:104]: i think saying the commit here adds confusion. it feels implied that we captured from head.

File: vscode/src/capture/command.ts (hunk 4 above)
Comment (Daanikus) [reply vscode/src/capture/command.ts:58]: This is the same as dash default

Comment (Daanikus) [reply vscode/src/capture/command.ts:58]: And no, it can't be null

File: vscode/src/capture/command.ts (hunk 5 above)
Comment (Daanikus) [reply vscode/src/capture/command.ts:104]: It's more for future reference. If the code changes and you go to revisit the rule (and haven't updated desc from default), you could be looking at irrelevant code and sit there wondering why it's "bad"

**Pull request merged**